pkg/controller/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -134,8 +135,10 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 // DeleteUserSignup deletes the specified UserSignup
 func (r *Reconciler) DeleteUserSignup(userSignup *toolchainv1alpha1.UserSignup, logger logr.Logger) error {
-
-	err := r.Client.Delete(context.TODO(), userSignup)
+	propagationPolicy := metav1.DeletePropagationForeground
```

Comment:
to test this option, you could use the mock client and set the `foregroundDeletion` finalizer on the UserSignup instead of deleting it, as the doc says:

> API sever will put the "foregroundDeletion" finalizer on the object, and sets its deletionTimestamp. This policy is cascading, i.e., the dependents will be deleted with Foreground.

it's not much as a test, but if someone in the future removes this option, the test will fail, so at least we have a (small) safety